### ISSUE TITLE
Caret error when inserting parenthesis around link

### ISSUE THREAD

**Issue by anandrajaram** [careteditor/caret#53]
The following error has occurred:
```
{
    "message": "Invalid count value",
    "version": "1.1.0",
    "platform": "darwin",
    "os": "15.3.0",
    "path": "web/Editor.Markdown.js",
    "code": "(_0x10b5x2[_0xfc65[70]](_0x10b5xe[_0xfc65[128]],_0x10b5xe[_0xfc65[130]]),_0x10b5x2[_0xfc65[74]](_0x10b5xe[_0xfc65[128]],_0x10b5xe[_0xfc65[119]][_0xfc65[85]](_0x10b5xe[_0xfc65[130]]+_0x10b5xe[_0xfc65[129]]-this[_0xfc65[11]])),_0x10b5x2[_0xfc65[75]](_0x10b5x3))},_deleteSetext:function()",
    "line": 5,
    "column": 19728,
    "stack": [
        "at String.repeat ((native)",
        "at Object.EditorMarkdown.(anonymous function)._resizeSetext (web/Editor.Markdown.js:5:19728)",
        "at Object.EditorMarkdown.(anonymous function)._onBeforeUpdate (web/Editor.Markdown.js:5:24544)",
        "at loopEvents (web/Events.js:5:229)",
        "at Object.Events.trigger (web/Events.js:5:527)",
        "at Object.Editor.(anonymous function)._update (web/Editor.js:5:13956)",
        "at Object.Editor.(anonymous function)._oninput (web/Editor.js:5:18701)"
    ]
}
```

**Comment by erusev** [comment]
To reproduce:

1. ![image](https://cloud.githubusercontent.com/assets/184170/13548658/fa810f98-e2fd-11e5-96c9-3beb98573e5d.png)

2. Hit <kbd>enter</kbd>

**Comment by erusev** [comment]
@astoilkov 

Here's one more scenario:

1. ![image](https://cloud.githubusercontent.com/assets/184170/13731563/ec563f60-e974-11e5-99ae-e2b16b047cf8.png)
2. Hit <kbd>Backspace</kbd>

**Comment by astoilkov** [comment]
This is not fixed in the 1.2.0 release.
